### PR TITLE
Get product api

### PR DESCRIPTION
**Pull request by mflinn92**
Feel free to take a look, got api set up for getting individual products and a few small adjustments to seeding script

### CODE REVIEW

File: server/models.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const mongoose = require('mongoose');
+const productDetailsModel = require ('../db/index.js').productDetails;
```
Comment (matt-leyrat) [inline server/models.js]: You can use destructuring here to clean this up.

File: server/models.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const mongoose = require('mongoose');
+const productDetailsModel = require ('../db/index.js').productDetails;
+
+class productDetails {
+  constructor() {
+    this.model = productDetailsModel;
+  }
+}
```
Comment (matt-leyrat) [inline server/models.js]: Don't forget to add new lines at the end of every file! (or have the linter do it for you)

File: server/models.js [hunk 3]
```diff
@@ -5,4 +5,11 @@ class productDetails {
   constructor() {
     this.model = productDetailsModel;
   }
-}
\ No newline at end of file
+
+  async getProduct(productId) {
```
Comment (matt-leyrat) [inline server/models.js]: The way you're using async/await is a little funky here

File: db/index.js [hunk 4]
```diff
@@ -28,4 +28,4 @@ const productDetailsSchema = new mongoose.Schema({
 
 const productDetails = mongoose.model('productDetails', productDetailsSchema);
 
-module.exports.productDetails = productDetails;
\ No newline at end of file
+module.exports.productDetails = productDetails;
```
Comment (matt-leyrat) [inline db/index.js]: There might be a more intuitive way to export this https://nodejs.org/api/modules.html

File: db/seed.js [hunk 5]
```diff
@@ -100,7 +100,7 @@ products = [];
 let product;
 for (let i = 0; i < 150; i++) {
   product = generator.generateProduct();
-  product.productId = i;
+  product.productId = i+1;
```
Comment (matt-leyrat) [inline db/seed.js]: You could also modify the for loop `for (let i = 1; i <= 150; i++)`

File: server/models.js (hunk 3 above)
Comment (mflinn92) [reply server/models.js]: This was the first time I tried using it, just wanted to try getting it to work. Out of curiosity what is funky about it.

Comment (matt-leyrat) [reply server/models.js]: I take it back, it looks fine 👍I wasn't sure how you implemented `findOne` but now that I have the context it makes sense :)

File: server/models.js [hunk 6]
```diff
@@ -5,4 +5,11 @@ class productDetails {
   constructor() {
     this.model = productDetailsModel;
   }
-}
\ No newline at end of file
+
+  async getProduct(productId) {
+    let productData = await productDetailsModel.findOne({productId: productId})
```
Comment (matt-leyrat) [inline server/models.js]: here's another opportunity to use destructuring

**Pull request merged**